File: js/euler86.js
function euler86(){
    let count = 0, res = 0;
    for (let a = 1; ; a++){
        for (let b = 1; b <= a; b++){
            for (let c = 1; c <= b; c++){
                let r = Math.sqrt(a * a + (b + c) * (b + c));
                if (r == Math.floor(r)) count++;
                if (count >= 1e6) return a;
            }
        }
    }
}

console.log(euler86());

File: go/euler86.go
package main
import (
    "fmt"
    "math"
)

func getMinDimension(minSols int) int {
    solutions := 0
    for a := 1; ; a++ {
        for b := 1; b <= a; b++ {
            for c := 1; c <= b; c++ {
                route := (a * a) + (b + c) * (b + c)
                s := int(math.Sqrt(float64(route)))
                if s * s == route {
                    solutions++
                    if solutions >= minSols {
                        return a
                    }
                }
            }
        }
    }
}

func main() {
    minSols := 1000000
    fmt.Println(getMinDimension(minSols))
}
